src/Filters/design.jl

```diff
@@ -310,23 +321,27 @@ end
 # The Octave implementation was not consulted in creating this code.
 
 # Create a lowpass filter from a lowpass filter prototype
-transform_prototype(ftype::Lowpass, proto::ZeroPoleGain) =
-    ZeroPoleGain(ftype.w * proto.z, ftype.w * proto.p,
+transform_prototype(ftype::Lowpass, proto::ZeroPoleGain{Z, P, K}) where {Z, P, K} =
+    ZeroPoleGain{Z, P, K}(ftype.w * proto.z, ftype.w * proto.p,
```

Comment:
So the type parameters will dictate the `Vector{Z}`, etc, type, as per the definition of a constructor. 
Also, the constructor is [tested here](https://github.com/JuliaDSP/DSP.jl/blob/50a03a299ae9d6b850466d3263013a31a3ce7643/test/filter_conversion.jl#L170) since the `convert` call translates to a constructor call which results in a `ZeroPoleGain{Z,P,K}(f.z, f.p, f.k)`, which forces a conversion.

You can test it out with a trivial example via:

```julia
struct A{T}
    x::T
end

A{Int}(2.0)
# A{Int64}(2)
```

Finally the test sets I added in `filter_design.jl:556` onwards explicitly has those tests.

Reply:
Ah you're totally right! I never knew that about constructors.